torchrl/objectives/costs/dreamer.py

```diff
@@ -0,0 +1,295 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from typing import Optional
+
+import torch
+
+from torchrl.data import TensorDict
+from torchrl.envs.model_based.dreamer import DreamerEnv
+from torchrl.envs.utils import set_exploration_mode
+from torchrl.envs.utils import step_tensordict
+from torchrl.modules import TensorDictModule
+from torchrl.objectives.costs.common import LossModule
+from torchrl.objectives.costs.utils import hold_out_net, distance_loss
+from torchrl.objectives.returns.functional import vec_td_lambda_return_estimate
+
+
+class DreamerModelLoss(LossModule):
+    """Dreamer Model Loss
+
+    Computes the loss of the dreamer world model. The loss is composed of the kl divergence between the prior and posterior of the RSSM,
+    the reconstruction loss between the observation and the reconstructed observation and the reward loss between the true reward and the predicted reward.
```

Comment:
can we reformulate `loss between X and Y`? Per se a loss is not a distance or a divergence...